pkg/controller.v1alpha3/experiment/experiment_controller.go

```diff
@@ -191,8 +191,9 @@ func (r *ReconcileExperiment) Reconcile(request reconcile.Request) (reconcile.Re
 		// Experiment is restartable only if it is in succeeded state by reaching max trials
 		if util.IsCompletedExperimentRestartable(instance) {
 			// Check if max trials is reconfigured
-			if (instance.Spec.MaxTrialCount != nil) &&
-				(*instance.Spec.MaxTrialCount != instance.Status.Trials) {
+			if (instance.Spec.MaxTrialCount != nil &&
+				*instance.Spec.MaxTrialCount != instance.Status.Trials) ||
+				(instance.Spec.MaxTrialCount == nil && instance.Status.Trials != 0) {
```

Comment:
should we consider the case: instance.Spec.MaxTrialCount == 0 and the experiment will be marked completed with instance.Status.Trials == 0; then update instance.Spec.MaxTrialCount to nil?

Reply:
i think, this is a invalid case. It doesn't make sense to set MaxTrialCount to be zero if set. We should add a validation for this separately. 
Related: https://github.com/kubeflow/katib/issues/768